### PR TITLE
Added auto complete address, added locate. Redesigned admin map. Added search on map.

### PR DESCRIPTION
**Pull request by Zairbekov**

### CODE REVIEW

File: src/app/pages/admin-page/forms/office-form/office-form.component.ts [hunk 1]
```diff
@@ -77,10 +77,26 @@ export class OfficeFormComponent implements OnInit {
...
+        const address = result.address;
+        dialogRef.afterClosed().subscribe(() => {
+          this.officeForm.setValue({
+            id: 101,
+            country: address.country,
+            city: address.city,
+            street: address.road,
+            house: address.house_number,
+            room: '',
+            phone: 123,
```
Comment (AlexanderBazukevich) [inline src/app/pages/admin-page/forms/office-form/office-form.component.ts:92]: We can skip form-fields, that we don't get from map like room, phone...

File: src/app/pages/home-page/offer-list-page/offer-list-page.component.html [hunk 2]
```diff
@@ -8,4 +8,7 @@
   >
     <app-offer-item [offer]="offer" class="offer-card"></app-offer-item
   ></a>
+  <div class="offers__text text_s_xl text_c_secondary" *ngIf="!offers?.length">
```
Comment (AlexanderBazukevich) [inline src/app/pages/home-page/offer-list-page/offer-list-page.component.html]: even if we use only modificators, we mustn`t skip block basic class. For this case - 'text'.

File: src/app/pages/home-page/office-item-page/office-item-page.component.html [hunk 3]
```diff
@@ -45,9 +45,14 @@
...
       ></mat-tab>
 
-      <mat-tab label="{{ 'description' | translate }}">{{
-        vendor.description || "Sorry, the brand didn't provide a description"
-      }}</mat-tab>
+      <mat-tab label="{{ 'description' | translate }}">
+        <div class="tab__item_indent">
+          {{
+            vendor.description ||
+              "Sorry, the brand didn't provide a description"
```
Comment (AlexanderBazukevich) [inline src/app/pages/home-page/office-item-page/office-item-page.component.html]: translate of this?

File: src/app/pages/home-page/vendor-item-page/vendor-item-page.component.html [hunk 4]
```diff
@@ -29,31 +29,38 @@
...
       </mat-tab>
 
-      <mat-tab label="{{ 'description' | translate }}">{{
-        vendor.description || "Sorry, the brand didn't provide a description"
-      }}</mat-tab>
+      <mat-tab label="{{ 'description' | translate }}">
+        <div class="tab__item_indent">
+          {{
+            vendor.description ||
+              "Sorry, the brand didn't provide a description"
```
Comment (AlexanderBazukevich) [inline src/app/pages/home-page/vendor-item-page/vendor-item-page.component.html]: no translate again

File: src/app/shared/custom-components/map/map-base/map-base.component.ts [hunk 5]
```diff
@@ -3,30 +3,69 @@ import {
...
+
+    this.map.on('moveend', () => {
+      //TODO: i will change all logic here, dont worry guys)
+      if (this.markers) {
+        for (const marker of this.markers) {
+          if (
+            marker._icon &&
+            !this.map.getBounds().contains(marker.getLatLng())
+          ) {
+            console.log(marker);
```
Comment (AlexanderBazukevich) [inline src/app/shared/custom-components/map/map-base/map-base.component.ts:62]: forgot?

**Review by AlexanderBazukevich — CHANGES_REQUESTED** [review]

File: src/app/pages/home-page/offer-list-page/offer-list-page.component.html (hunk 2 above)
Comment (abarmina) [reply src/app/pages/home-page/offer-list-page/offer-list-page.component.html]: it's my mistake, I'll fix it

File: src/app/pages/home-page/office-item-page/office-item-page.component.html (hunk 3 above)
Comment (abarmina) [reply src/app/pages/home-page/office-item-page/office-item-page.component.html]: we have! I'll fix it

File: src/app/pages/home-page/vendor-item-page/vendor-item-page.component.html (hunk 4 above)
Comment (abarmina) [reply src/app/pages/home-page/vendor-item-page/vendor-item-page.component.html]: ok, I'll fix it

**Review by tlazarchuk — APPROVED** [review]

**Review by AlexanderBazukevich — APPROVED** [review]

**Pull request merged**